doc/amoc_coordinator.md

```diff
@@ -0,0 +1,167 @@
+When the `Number Of Users` is set to a positive integer, all `Coordination Actions` with arities `/1, /2` and `/3` are handled.
+A new batch size is set in the `NumberOfUsers`. Each user in the batch calls the `add` function registering to the coordinator and triggering the `Coordination Plan`.
+If more then one of the `Coordination Items` matching the `NumberOfUsers` is triggered, each of them will be passed a respective number of users.
+For example if the `Coordination Plan` is `[{2, Act1}, {3, Act2}]` then on the 6th user calling `add`, `Act1` will be called with 2 users passed and `Act2` wil be called with 3 users passed.
+
+`Coordination Actions` may be one of the following:
+ - `fun(Event) -> any()` - this type of action does not care about particular users, but only about the number of them;
+ - `fun(Event, ListOfUsersData) -> any()` - this type of action gets `ListOfUsersData` which is a list of tuples `{Pid, Data}` with `Pid`s passed by users calling `amoc_coordinator:add/2` or `amoc_coordinator:add/3`;
+- `fun(Event, User1, User2) -> any()` - this type of action gets `distinct pairs` from the batch of users `User1` and `User2` which are tuples `{Pid, Data}` with `Pid`s passed by users calling `amoc_coordinator:add/2` or `amoc_coordinator:add/3`;
+
+`Coordination Actions` are executed in reversed order in relation to thair declaration in the `Coordination Plan`.
+
+`distinct pairs` - in the context, these are pairs from collection where:
+ - when `{A, B}` is in the `distinct pairs` then `{B, A}` is not;
+ - when `{A, A}` is not in the `distinct pairs`;
+ - all pairs are distinct;
+ - Eg. for `[a]` `distinct pairs` collection is `[{a, undefined}]`;
+ - Eg. for `[a, b]` `distinct pairs` collection is `[{a, b}]`;
+ - Eg. for `[a, b, c]` `distinct pairs` collection is `[{a, b}, {a, c}, {b, c}]`.
+
+## Exports
+
+#### `start(CoordinatorName, CoordinationPlan) -> ok | error`
+#### `start(CoordinatorName, CoordinationPlan, Timeout) -> ok | error`
+
+##### Types
+```erlang
+CoordinatorName :: atom()
+CoordinationPlan  :: [ CoordinationItem ]
+CoordinationItem :: {NumberOfUsers :: pos_integer() | all, CoordinationActions}
+CoordinationActions :: [ CoordinationActions ] | CoordinationAction
+CoordinationAction :: 
+    fun(Event) -> any() |
+    fun(Event, ListOfUsersData) -> any() |
+    fun(Event, MaybeUser, MaybeUser) -> any()
+Event :: coordinate | timeout | stop | reset
+ListOfUsersData :: [ User ]
+MaybeUser :: undefined | User
+User :: {UsersPid :: pid(), Data :: any()}
+Timeout :: pos_integer() | infinity
+```
+
+This function starts a coordinator. Usually is called in `init/0` of a amoc scenario.
+
+#### `add(CoordinatorName, Data) -> ok`
+#### `add(CoordinatorName, Pid, Data) -> ok`
+
+##### Types
+```erlang
+CoordinatorName :: atom()
+UsersPid :: pid()
+Data :: any()
+```
+
+This function adds current process data. Usually is called in `start/2` of a amoc scenario.
```

Comment:
I think it's important to say that this function basically increases number of already started users. Without this the coordination actions may not be triggered, especially the one for all users. If for some reason one session doesn't call the function and we wanted to have 10 users in the test, then the rule for `all` will not be triggered. Am I right?

Reply:
No

Reply:
If we have 10 users in a test but only 5 of them calls the function `add`. and the event `timeout` happens action for `all` will be triggered for the 5 users who registered to the amoc coordinator.